sdk/include/platform/sunburst/platform-i2c.hh

```diff
@@ -0,0 +1,352 @@
+	// I2C clock stretching timeout control.
+	uint32_t timeoutCtrl;
+	// I2C target address and mask pairs
+	uint32_t targetId;
+	// I2C target acquired data
+	uint32_t acqdata;
+	// I2C target transmit data
+	uint32_t txdata;
+	// I2C host clock generation timeout value (in units of input clock
+	// frequency).
+	uint32_t hostTimeoutCtrl;
+	// I2C target internal stretching timeout control.
+	uint32_t targetTimeoutCtrl;
+	// Number of times the I2C target has NACK'ed a new transaction since the
+	// last read of this register.
+	uint32_t targetNackCount;
+	// Timeout in Host-Mode for an unhandled NACK before hardware automatically
+	// ends the transaction.
+	uint32_t targetAckCtrl;
+
+	// Status Register Values
+	enum : uint32_t
+	{
+		// Host mode FMT FIFO is full
+		StatusFmtFull = 1 << 0,
+		// Host mode RX FIFO is full
+		StatusRxFull = 1 << 1,
+		// Host mode FMT FIFO is empty
+		StatusFmtEmpty = 1 << 2,
+		// Host functionality is idle. No Host transaction is in progress
+		StatusHostIdle = 1 << 3,
+		// Target functionality is idle. No Target transaction is in progress
+		StatusTargetIdle = 1 << 4,
+		// Host mode RX FIFO is empty
+		StatusRxEmpty = 1 << 5,
+		// Target mode TX FIFO is full
+		StatusTxFull = 1 << 6,
+		// Target mode receive FIFO is full
+		StatusAcqFull = 1 << 7,
+		// Target mode TX FIFO is empty
+		StatusTxEmpty = 1 << 8,
+		// Target mode receive FIFO is empty
+		StatusAcqEmpty = 1 << 9,
+		// Target mode stretching at (N)ACK phase due to zero count in
+		// TARGET_ACK_CTRL.NBYTES
+		StatusAckCtrlStretch = 1 << 10,
+	};
+
+	enum : uint32_t
+	{
+		// Issue a START condition before transmitting BYTE.
+		FDataStart = 1 << 8,
+		// Issue a STOP condition after this operation
+		FDataStop = 1 << 9,
+		// Read BYTE bytes from I2C. (256 if BYTE==0)
+		FDataReadB = 1 << 10,
+		// Do not NACK the last byte read, let the read
+		// operation continue
+		FDataRCont = 1 << 11,
+		// Do not signal an exception if the current byte is not ACK’d
+		FDataNAkOk = 1 << 12,
+	};
+
+	enum : uint32_t
+	{
+		// RX fifo reset. Write 1 to the register resets RX_
+		FifoCtrlRxRst = 1 << 0,
+		// FMT fifo reset. Write 1 to the register resets FMT_FIFO. Read returns
+		// 0
+		FifoCtrlFmtRst = 1 << 1,
+		// ACQ FIFO reset. Write 1 to the register resets it. Read returns 0
+		FifoCtrlAcqRst = 1 << 7,
+		// TX FIFO reset. Write 1 to the register resets it. Read returns 0
+		FifoCtrlTxRst = 1 << 8,
+	};
+
+	void reset_fifos() volatile
+	{
+		fifoCtrl =
+		  (FifoCtrlRxRst | FifoCtrlFmtRst | FifoCtrlAcqRst | FifoCtrlTxRst);
+	}
+
+	void set_host_mode() volatile
+	{
+		ctrl = 1;
+	}
+
+	/**
+	 * Set the I2C timing parameters appropriately for the given bit rate.
+	 * Distilled from:
+	 * https://opentitan.org/book/hw/ip/i2c/doc/programmers_guide.html
+	 */
+	void set_speed(const uint32_t SpeedKhz) volatile
+	{
+		// We must round up the system clock frequency to lengthen intervals.
+		constexpr uint32_t SysclkKhz = (I2cDevClockHz + 999) / 1000;
```

Comment:
Use `round_up_divide` here?

Reply:
This is a 32-bit value, round_up_divide truncates to a 16-bit value